src/test/java/unit/GcsBucketReferenced.java

```diff
@@ -345,5 +391,143 @@ void updateMultipleOrNoProperties() throws IOException {
         TestCommand.runAndParseCommandExpectSuccess(
             UFGcsBucket.class, "resource", "describe", "--name=" + newName);
     assertEquals(newDescription, describeBucket.description);
+    assertEquals(externalPrivateBucket.getName(), describeBucket.bucketName);
+  }
+
+  @Test
+  @DisplayName(
+      "Attempt to update the gcs bucket while the user only have access to the externalSharedBucket")
+  void updateGcsBucketWithPartialAccess() throws IOException {
+    workspaceCreator.login();
+
+    UFWorkspace createWorkspace =
+        TestCommand.runAndParseCommandExpectSuccess(UFWorkspace.class, "workspace", "create");
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+    // `terra workspace add-user --email=$email --role=READER`
+    TestCommand.runCommandExpectSuccess(
+        "workspace", "add-user", "--email=" + shareeUser.email, "--role=WRITER");
+
+    String name = "updateGcsBucketWithPartialAccess";
+    TestCommand.runCommandExpectSuccess(
+        "resource",
+        "add-ref",
+        "gcs-bucket",
+        "--name=" + name,
+        "--bucket-name=" + externalSharedBucket.getName());
+
+    shareeUser.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+
+    String newName = "updateGcsBucketWithPartialAccess_NEW";
+    UFGcsBucket updateBucket =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class,
+            "resource",
+            "update",
+            "gcs-bucket",
+            "--name=" + name,
+            "--new-name=" + newName);
+    assertEquals(newName, updateBucket.name);
+
+    TestCommand.runCommandExpectExitCode(
+        1,
```

Comment:
does this fail because the user is a READER or because they don't have access to the new bucket?